### PR TITLE
Replace usage of deprecated ConfigProperties with ConfigMapping

### PR DESCRIPTION
**Pull request by nastra**
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1814)

### CODE REVIEW

File: servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusServerConfig.java [hunk 1]
```diff
@@ -16,18 +16,21 @@
 package org.projectnessie.server.config;
 
 import io.quarkus.arc.config.ConfigProperties;
-import org.eclipse.microprofile.config.inject.ConfigProperty;
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithName;
 import org.projectnessie.services.config.ServerConfig;
 
 /** Nessie server config for Quarkus. */
 @ConfigProperties(prefix = "nessie.server")
```
Comment (nastra) [inline servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusServerConfig.java:24]: switching this one currently still fails and I need to investigate why

**Review by dimas-b — APPROVED** [review]

File: servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusServerConfig.java (hunk 1 above)
Comment (dimas-b) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusServerConfig.java]: Did we have to upgrade Quarkus to fix this?

**Review by dimas-b — APPROVED** [review]

File: servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusServerConfig.java (hunk 1 above)
Comment (nastra) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusServerConfig.java]: yep, it was wrongly marked in their release notes that it's fixed in 2.1.3.Final, but it was actually fixed in 2.2.0.CR1

File: pom.xml [hunk 2]
```diff
@@ -155,7 +155,7 @@
     <openapi.version>2.0</openapi.version>
     <prometheus.version>0.9.0</prometheus.version>
     <protobuf.version>3.17.3</protobuf.version>
-    <quarkus.version>2.1.3.Final</quarkus.version>
+    <quarkus.version>2.2.0.CR1</quarkus.version>
```
Comment (nastra) [inline pom.xml:158]: `2.2.0.Final` should be released today, so we/dependabot can switch to that new version then

**Pull request merged**